### ISSUE TITLE
the trait `Clone` is not implemented for `Vector<KeyPoint>`

### ISSUE THREAD

**Issue by mdenty** [twistedfall/opencv-rust#426]
1. Operating system : macOS 13.2.1
2. The way you installed OpenCV: Homebrew
3. OpenCV version 4.7.0
4. rustc version (`rustc --version`) : rustc 1.67.1 (d5a82bbd2 2023-02-07)
5. Attach the full output of the following command from your project directory:
   ```shell script
   RUST_BACKTRACE=full cargo build -vv 
   ```
[Here is the compile log](https://github.com/twistedfall/opencv-rust/files/10744604/compile.txt)

The code where the unexpected error appears.
It looks like a regression.

```rust
#[derive(Clone)]
pub(crate) struct FeaturesModel {
    model: Model,
    kp_model: Vector<KeyPoint>,
    descriptors: Mat,
    model_width: i32,
    model_height: i32,
}
```

I'm upgrading from 0.72.1 and my project does not compile anymore.

For the record, I was trying to upgrade since `opencv::dnn::Net::forward` is crashing always in 0.72.1.

```rust
fn RGB(r: f64, g: f64, b: f64) -> Scalar {
    Scalar::new(b, g, r, 0.0)
}

fn detect_face(
    scene: &Mat,
    face_models: &mut Vec<FaceModel>,
) -> Result<FaceDetectionResult, opencv::Error> {
    assert_eq!(scene.channels(), 3);
    for face_model in face_models {
        let mut net = face_model.manager.load_model().unwrap();
        net.set_preferable_backend(DNN_BACKEND_OPENCV)?;
        net.set_preferable_target(DNN_TARGET_CPU)?;

        println!("build: {}", get_build_information()?);
        let size = Size::new(300, 300);
        let mean = RGB(104.0, 177.0, 123.0);
        let mut resized = Mat::default();
        let size0 = scene.size().unwrap();
        let w = size0.width;
        let h = size0.height;
        resize(&scene, &mut resized, size, 0.0, 0.0, INTER_LANCZOS4)?;
        imwrite("resized.png", &resized, &Vector::default())?;
        if let Ok(blob) = blob_from_image(&resized, 1.0, size, mean, false, false, CV_32F) {

            net.set_input(&blob, "", 1.0, Scalar::default())?;

            let mut output_blobs: Vector<Mat> = Vector::new();
            let out_blob_names = Vector::new();
            net.forward(&mut output_blobs, &out_blob_names)?; // <-- crash here
            // for r in 0..output_blobs.rows() {
            //     for c in 0..output_blobs.cols() {
            //         println!("at: {c}, {r} --> {:?}", output_blobs.at_2d::<i32>(r, c))
            //     }
            // }
        }
    }
    todo!()
}
```

produces this crash log:

```log
Thread 0 Crashed:: main Dispatch queue: com.apple.main-thread
0   libopencv_dnn.4.7.0.dylib     	       0x1028641a4 cv::dnn::dnn4_v20221220::Net::Impl::forward(cv::_OutputArray const&, std::__1::vector<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, std::__1::allocator<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > > > const&) + 400
1   libopencv_dnn.4.7.0.dylib     	       0x102864158 cv::dnn::dnn4_v20221220::Net::Impl::forward(cv::_OutputArray const&, std::__1::vector<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, std::__1::allocator<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > > > const&) + 324
2   libopencv_dnn.4.7.0.dylib     	       0x10285afb0 cv::dnn::dnn4_v20221220::Net::forward(cv::_OutputArray const&, std::__1::vector<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, std::__1::allocator<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > > > const&) + 84
3   the-gr-project                	       0x100241d2c cv_dnn_Net_forward_const__OutputArrayR_const_vector_String_R + 44
4   the-gr-project                	       0x1000e5c54 opencv::opencv::hub::dnn::NetTrait::forward::h0276ac933f08810b + 364 (dnn.rs:8915)
5   the-gr-project                	       0x100107368 the_gr_project::features_matcher::detect_face::h74a8bc1a292bd0ea + 2124 (features_matcher.rs:405)
6   the-gr-project                	       0x100105570 the_gr_project::features_matcher::Finder::detect_file::h6a60ef850d44c7f2 + 1720 (features_matcher.rs:216)
7   the-gr-project                	       0x1000fd808 the_gr_project::main::hc8655a7ae32c75ce + 3400 (main.rs:130)
8   the-gr-project                	       0x100107b2c core::ops::function::FnOnce::call_once::h41a0a66d2b66c2bc + 20 (function.rs:507)
9   the-gr-project                	       0x1001209b0 std::sys_common::backtrace::__rust_begin_short_backtrace::h396a0e6f651b5e3a + 24 (backtrace.rs:121)
10  the-gr-project                	       0x1000e1f54 std::rt::lang_start::_$u7b$$u7b$closure$u7d$$u7d$::h57d6a3c8d3f12598 + 28 (rt.rs:166)
11  the-gr-project                	       0x1002585ec std::rt::lang_start_internal::h00a235e820a7f01c + 616
12  the-gr-project                	       0x1000e1f20 std::rt::lang_start::he2f67473492ea5b0 + 84 (rt.rs:165)
13  the-gr-project                	       0x1000fe40c main + 36
14  dyld                          	       0x1816dfe50 start + 2544
```

The original crash with `forward` is now worked around by using :
```rust 
let mat = net.forward_single("")?;
```
but I still wonder why it crashes with the other form of `forward`

**Comment by twistedfall** [comment]
Hi, thanks for the report! I can confirm the regression, I'll see what can be done to fix that. Regarding the crash, can you please provide the complete minimum reproducible example that I can use to test?

**Comment by mdenty** [comment]
@twistedfall: Sure, I've attached a [sample.zip](https://github.com/twistedfall/opencv-rust/files/10767248/sample.zip) code with the model from https://github.com/gopinath-balu/computer_vision/tree/master/CAFFE_DNN.
I also post the code here for readability.

I suspect that the crash comes from the parameters to `forward()`, I did not found a way to specify them correctly.
 

```rust
use opencv::core::{scale_add, Scalar, Size, Vector, CV_32F};
use opencv::dnn::{blob_from_image, Net};
use opencv::imgcodecs::{imread, IMREAD_ANYCOLOR};
use opencv::imgproc::{cvt_color, resize, COLOR_BGRA2BGR, COLOR_GRAY2BGR, INTER_LANCZOS4};
use opencv::prelude::{Mat, MatTraitConst, NetTrait};
use std::env;
use std::process::exit;

fn main() -> Result<(), opencv::Error> {
    let args: Vec<String> = env::args().collect();
    if args.len() < 2 {
        println!("usage: {} <image_to_test>", args[0]);
        exit(2);
    }
    println!("Hello, world: try to detect face on {}!", args[1]);
    let mat = imread(args[1].as_str(), IMREAD_ANYCOLOR)?;
    if mat.empty() {
        println!("Nothing loaded: is {} there ?", args[1]);
        exit(3);
    }
    // detect_face(mat, false);
    detect_face(mat, true)
}

pub fn rgb(r: f64, g: f64, b: f64) -> Scalar {
    Scalar::new(b, g, r, 0.0)
}

fn load_net() -> Result<Net, opencv::Error> {
    let prototxt = "cnn/deploy.prototxt.txt";
    let caffe_model = "cnn/res10_300x300_ssd_iter_140000.caffemodel";
    opencv::dnn::read_net_from_caffe(prototxt, caffe_model)
}

fn detect_face(mat: Mat, will_crash: bool) -> Result<(), opencv::Error> {
    let scene = if mat.channels() != 3 {
        let mut dst = Mat::default();
        let action = if mat.channels() == 4 {
            COLOR_BGRA2BGR
        } else {
            COLOR_GRAY2BGR
        };
        cvt_color(&mat, &mut dst, action, 0)?;
        dst
    } else {
        mat
    };

    let size = Size::new(300, 300);
    let mean = rgb(104.0, 177.0, 123.0);
    let mut resized = Mat::default();
    resize(&scene, &mut resized, size, 0.0, 0.0, INTER_LANCZOS4)?;

    if let Ok(blob) = blob_from_image(&resized, 1.0, size, mean, false, false, CV_32F) {
        let mut net = load_net()?;
        net.set_input(&blob, "", 1.0, Scalar::default())?;
        if will_crash {
            // does not compile (see: https://docs.opencv.org/4.7.0/db/d30/classcv_1_1dnn_1_1Net.html#a00e707a86b2da4f980f9342b1fc2cc92)
            // let mut output_blobs: Vector<Vector<Mat>> = Vector::new();
            let mut output_blobs: Vector<Mat> = Vector::new();
            let out_blob_names = Vector::new();

            // Thread 0 Crashed:: main Dispatch queue: com.apple.main-thread
            // 0   libopencv_dnn.4.7.0.dylib     	       0x104df41a4 cv::dnn::dnn4_v20221220::Net::Impl::forward(cv::_OutputArray const&, std::__1::vector<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, std::__1::allocator<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > > > const&) + 400
            // 1   libopencv_dnn.4.7.0.dylib     	       0x104df4158 cv::dnn::dnn4_v20221220::Net::Impl::forward(cv::_OutputArray const&, std::__1::vector<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, std::__1::allocator<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > > > const&) + 324
            // 2   libopencv_dnn.4.7.0.dylib     	       0x104deafb0 cv::dnn::dnn4_v20221220::Net::forward(cv::_OutputArray const&, std::__1::vector<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, std::__1::allocator<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > > > const&) + 84
            // 3   minimal-crash                 	       0x102b1eedc cv_dnn_Net_forward_const__OutputArrayR_const_vectorLStringGR + 44
            // 4   minimal-crash                 	       0x102b15ddc opencv::opencv::hub::dnn::NetTrait::forward::h0784ae3d7e62a678 + 364 (dnn.rs:8956)
            // 5   minimal-crash                 	       0x102b16d64 minimal_crash::detect_face::h1ffc935f9ff0e6ba + 1576 (main.rs:63)

            net.forward(&mut output_blobs, &out_blob_names)?;
        } else {
            let mat = net.forward_single("")?;
            let dims = mat.dims();
            let mat_size = mat.mat_size();
            for i in 0..dims {
                println!("Dim {i}: {}", mat_size[i as usize])
            }
            let s = mat_size[2];
            println!("results count: {s}")
        }
    }

    Ok(())
}
```

**Comment by twistedfall** [comment]
The `Clone` not implemented problem has been fixed in version 0.78.1
